sync/src/blocks.rs

```diff
@@ -295,6 +295,10 @@ impl BlockCollection {
 		let old_subchains: HashSet<_> = { self.heads.iter().cloned().collect() };
 		for s in self.heads.drain(..) {
 			let mut h = s.clone();
+			if !self.blocks.contains_key(&h) {
```

Comment:
Fixed a bug here: If a header is inserted with a number immediately following the chain head it would get updated incorrectly, which led to same blocks requested twice.